src/components/Confirm/Confirm.js

```diff
@@ -220,10 +220,18 @@ class Confirm extends Component {
           : {}),
       }
       const issuingCountry = this.getIssuingCountry()
-      // API does not support 'residence_permit' type but does accept 'unknown'
-      // See https://documentation.onfido.com/#document-types
+
+      // Make sure documents always have a filename
+      // A `filename` might have been defined when the capture is created
+      // if filename is not present, check if `blob` has a property `name` (only available for File types, which come from the html5 file picker)
+      // alternatively use default filename
+      //
+      const blobName =
+        filename || blob.name || `document_capture.${mimeType(blob)}`
```

Comment:
Reading that comment above, it sounds like `name` property is not guaranteed in `blob` object. Probably best to use optional chaining here then.
```suggestion
        filename || blob?.name || `document_capture.${mimeType(blob)}`
```